test/e2e/e2e_test.go

```diff
@@ -0,0 +1,150 @@
+	thresholds[v1.ResourceMemory] = 20
+	thresholds[v1.ResourcePods] = 20
+	thresholds[v1.ResourceCPU] = 60
+	targetThresholds[v1.ResourceMemory] = 20
+	targetThresholds[v1.ResourcePods] = 20
+	targetThresholds[v1.ResourceCPU] = 70
+	// Run descheduler.
+	evictionPolicyGroupVersion, err := eutils.SupportEviction(clientset)
+	if err != nil || len(evictionPolicyGroupVersion) == 0 {
+		glog.Fatalf("%v", err)
+	}
+	stopChannel := make(chan struct{})
+	nodes, err := nodeutil.ReadyNodes(clientset, "", stopChannel)
+	if err != nil {
+		glog.Fatalf("%v", err)
+	}
+	nodeUtilizationThresholds := deschedulerapi.NodeResourceUtilizationThresholds{Thresholds: thresholds, TargetThresholds: targetThresholds}
+	nodeUtilizationStrategyParams := deschedulerapi.StrategyParameters{NodeResourceUtilizationThresholds: nodeUtilizationThresholds}
+	lowNodeUtilizationStrategy := deschedulerapi.DeschedulerStrategy{Enabled: true, Params: nodeUtilizationStrategyParams}
+	ds := &options.DeschedulerServer{Client: clientset}
+	strategies.LowNodeUtilization(ds, lowNodeUtilizationStrategy, evictionPolicyGroupVersion, nodes)
+	time.Sleep(10 * time.Second)
+
+	return
+}
+
+func podScheduled(c clientset.Interface, podNamespace, podName string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.Core().Pods(podNamespace).Get(podName, metav1.GetOptions{})
+		if err != nil {
+			// This could be a connection error so we want to retry.
+			return false, nil
+		}
+		if pod.Spec.NodeName == "" {
+			return false, nil
+		}
+		return true, nil
+	}
+}
+
+// waitForPodToScheduleWithTimeout waits for a pod to get scheduled and returns
+// an error if it does not scheduled within the given timeout.
+func waitForPodToScheduleWithTimeout(cs clientset.Interface, pod *v1.Pod, timeout time.Duration) error {
+	return wait.Poll(time.Second, timeout, podScheduled(cs, pod.Namespace, pod.Name))
+}
+
+func TestE2E(t *testing.T) {
+	// If we have reached here, it means cluster would have been already setup and the kubeconfig file should
+	// be in /tmp directory.
+	clientSet, _ := client.CreateClient("/tmp/admin.conf")
+	nodeList, _ := clientSet.Core().Nodes().List(metav1.ListOptions{})
+	// We may have a 3 node cluster by now. Let us mark one node as unschedulable so that, we can actually
+	// make sure that one node is heavily utilized.
+	for _, node := range nodeList.Items {
+		if len(node.Spec.Taints) == 0 && node.Spec.Unschedulable == false {
+			node.Spec.Unschedulable = true
+			break
+		}
+	}
+	podList := generatePods(clientSet, 4)
+	// This is to ensure that all pods got scheduled.
+	for _, pod := range podList {
+		waitForPodToScheduleWithTimeout(clientSet, pod, time.Second)
+	}
+	// Marking the node which was unschedulable earlier to schedulable again, so that pods
+	// could be moved to this node.
+	for _, node := range nodeList.Items {
+		if len(node.Spec.Taints) == 0 && node.Spec.Unschedulable == true {
+			node.Spec.Unschedulable = false
```

Comment:
Should not the code call Patch at least so the change takes effect in the cluster?

Reply:
AFAIK no, because we are partially updating the node object in the client side and we are operating on the node list again on the client side.